### PR TITLE
pointcnn_windows_guide

### PR DESCRIPTION
**Pull request by vbhv14**
latest guide as per new release. 
https://github.com/ArcGIS/geosaurus/issues/3848
&
https://github.com/ArcGIS/geosaurus/issues/3603

changes as compared to old guide:
- [x]  version updated for PyTorch, PyTorch-Geometric & fastai.
- [x]  fixed command for transforms3d, gdal, laspy.
- [x]  Additional out of conda, installations  are not required now (build_tools & cuda_toolkit setup). 
- [x]  separate instructions for pro & anaconda user. (to make the guide more robust.)
- [x]  added instructions for complete jupyter lab compatibility, for ARCGIS API for python.  
- [x] fixed 'out of memory' issues with building jupyter lab extensions.

### CODE REVIEW

**Comment by vbhv14** [comment]
cc @divyanshj16

**Comment by priyankatuteja** [comment]
@SurajBaloni Could you please help in reviewing this.

File: guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb [hunk 1]
```diff
@@ -18,7 +18,7 @@
    "cell_type": "markdown",
```
Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <ol><li>producing highly accurate<s> x,y,z</s> <strong>X, Y and Z</strong> measurements</li><li>which <s>include x, y, and z,</s> <strong>includes X, Y, and Z measurements</strong> along with additional attributes such as GPS <s>time stamps,</s> <strong>timestamps,</strong></li><li>intensity –&nbsp;the return strength of the laser pulse that generated the Lidar point, <s>and, number of returns – the total number of returns for a given pulse</s> <strong>and the total number of returns for a given pulse.</strong></li><li>Additionally, with further post<strong>-</strong>processing</li><li>Though the <s>PoinCNN</s> <strong>PointCNN </strong>network emulates traditional convolution neural networks</li><li><s>Point Cloud</s> Point cloud</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <p>but at the same<strong> time,</strong> they<strong> </strong>are sensor</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <ol><li>Left <s>Image</s> <strong>image </strong>shows a raw XYZ lidar points whereas <strong>the</strong> right image shows the same data labelled&nbsp;</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <ol><li>As a result, it's recommended to use either a labeled portion of the data <s> we are planning to do prediction on,</s>  <strong>on which we are planning to do the prediction or</strong> to use a more generalized training model</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <ol><li> After these <s>operations</s> <strong>operations,</strong> we get an array of points of the shape (P, K, 3+E)</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <ol><li>X-Conv is recursively applied to<s> “project”, or “aggregate”</s>, <strong>“project” or “aggregate”</strong> information <s>from neighborhoods</s> <strong>from the neighborhood</strong></li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <ol><li>To state it succinctly, PointCNN differs from conventional <s>grid based</s> <strong>grid-based</strong> CNNs primarily due to the application of <s>X-Conv layers;</s> <strong>X-Conv layers.</strong> <strong>E</strong>ven then, the general process is</li><li><strong>T</strong>he way the local regions are extracted</li><li><strong>T</strong>he way the information from local regions is learned</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <p>In each X-Conv operation N represents <strong>the </strong>number of <strong>the </strong>points in next layer, C represents <strong>the </strong>number of channels, K represents the number of nearest neighbors and D represents <strong>the </strong>dilation rate.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <p>Implementation <s>in</s> <strong>using </strong>arcgis.learn <strong>module</strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (SurajBaloni) [inline guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb:18]: <ol><li>When training a PointCNN model using <code>arcgis.learn </code>, the raw point cloud dataset in LAS files is <s>first</s> converted into blocks of points <s>containing </s> <strong>that contains</strong> a specific number of points and their labels</li><li>We can also specify all the additional parameters while initializing PointCNN <s>in the form of</s> <strong>by </strong>using keyword arguments. The keyword argument takes in a parameter named&nbsp;<code>encoder_params</code>&nbsp;<strong>,</strong><s>keyword argument, in </s> which <s>we can be</s> <strong>is </strong>used to specify the number of output channels for each layer in the encoder</li><li>There are mainly four things that we need to define in the <s>encoder,  </s><strong>encoder i.e.</strong><code>out_channels</code>, P, K and D.</li><li>The <code>out_channel</code> specifies <strong>the </strong>number of channels after each layer</li><li><code>P</code>, <code>K</code> and <code>D</code> <s>specifiy </s> <strong>specify</strong> the number of points, k-nearest-neighbor and <strong>the </strong>dilation rate <strong>respectively </strong>in each layer</li><li>Additionally, one can specify <code>m</code> <strong>which</strong> is the multiplicative <s>factor, which</s> factor and is multiplied by each element of <strong>the</strong> out_channel list</li><li>The <code>sample_point_num</code> keyword argument specifies the number of points that we would like to train the model. <s>on</s></li><li>maximum points in a block <s>during </s>while exporting </li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: ''corrected accordingly'', Thanks.

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: ''corrected accordingly'', Thanks.

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: ''corrected accordingly'', Thanks.

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: ''corrected accordingly'', Thanks.

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: ''corrected accordingly'', Thanks.

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: ''corrected accordingly'', Thanks.

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: ''corrected accordingly'', Thanks.

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: <p>meaning is conveyed, anyways.</p>

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: <p>''corrected accordingly'', Thanks.</p>

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: <p>''corrected accordingly'', Thanks.</p>

Comment (SurajBaloni) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: <p>You can consider including:</p><p>1) Point clouds are large collections of 3D elevation points, which include x, y, and z <strong>measurements</strong></p><p>2) <s>post -processing</s> <strong>post-processing</strong>  (extra space in between)</p>

Comment (SurajBaloni) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: <p>You can consider including:</p><p>1) <s>grid -based</s>  <strong>grid-based</strong> (extra space in between)</p>

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: no extra spaces, were found, in HTML or notebook, might be due to ReviewNB.

Comment (vbhv14) [reply guide/14-deep-learning/point_cloud_segmentation_using_pointcnn.ipynb]: no extra space, were found, in HTML or notebook, might be due to ReviewNB.

**Comment by vbhv14** [comment]
<p>not found.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by vbhv14** [comment]
<p>not found</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by vbhv14** [comment]
<p>done.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by vbhv14** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/713/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by priyankatuteja** [comment]
@vbhv14 The change you added in the last commit it grammatically incorrect. Could you please revert back the last commit.

**Pull request merged**